docs/examples/tensorflow/demo/nvutils/cmdline.py

```diff
@@ -0,0 +1,130 @@
+    return cmdline
+
+
+class RequireInCmdline(object):
+    pass
+
+def _default(vals, key):
+    v = vals.get(key)
+    return None if v is RequireInCmdline else v
+
+def _required(vals, key):
+    return vals.get(key) is RequireInCmdline
+
+def parse_cmdline(init_vals, custom_parser=None):
+    if custom_parser is None:
+        f = argparse.ArgumentDefaultsHelpFormatter
+        p = argparse.ArgumentParser(formatter_class=f)
+    else:
+        p = custom_parser
+
+    p.add_argument('--data_dir',
+                   default=_default(init_vals, 'data_dir'),
+                   required=_required(init_vals, 'data_dir'),
+                   help="""Path to dataset in TFRecord format
+                   (aka Example protobufs). Files should be
+                   named 'train-*' and 'validation-*'.""")
+    p.add_argument('--data_idx_dir',
+                   default=_default(init_vals, 'data_idx_dir'),
+                   required=_required(init_vals, 'data_idx_dir'),
+                   help="""Path to index files of TFRecord dataset
+                   Files should be named 'train-*.idx' and
+                   'validation-*.idx'.""")
+    p.add_argument('-b', '--batch_size', type=int,
+                   default=_default(init_vals, 'batch_size'),
+                   required=_required(init_vals, 'batch_size'),
+                   help="""Size of each minibatch.""")
+    p.add_argument('-i', '--num_iter', type=int,
+                   default=_default(init_vals, 'num_iter'),
+                   required=_required(init_vals, 'num_iter'),
+                   help="""Number of batches or epochs to run.""")
+    p.add_argument('-u', '--iter_unit', choices=['epoch', 'batch'],
+                   default=_default(init_vals, 'iter_unit'),
+                   required=_required(init_vals, 'iter_unit'),
+                  help="""Select whether 'num_iter' is interpreted
+                  in terms of batches or epochs.""")
+    p.add_argument('--log_dir',
+                   default=_default(init_vals, 'log_dir'),
+                   required=_required(init_vals, 'log_dir'),
+                   help="""Directory in which to write training
+                   summaries and checkpoints.""")
+    p.add_argument('--display_every', type=int,
+                   default=_default(init_vals, 'display_every'),
+                   required=_required(init_vals, 'display_every'),
+                   help="""How often (in batches) to print out
+                   running information.""")
+    p.add_argument('--precision', choices=['fp32', 'fp16'],
+                   default=_default(init_vals, 'precision'),
+                   required=_required(init_vals, 'precision'),
+                   help="""Select single or half precision arithmetic.""")
+    p.add_argument('--dali_cpu', action='store_true',
+                   default=False,
+                   help="""Use CPU decoder for DALI for input pipeline""")
```

Comment:
I would say it is not only decoder but most of the pipeline is CPU based.

Reply:
Done